diff -r 3482d151136b src/edit.c
--- a/src/edit.c	Wed Apr 25 19:07:41 2012 +0200
+++ b/src/edit.c	Mon May 07 16:04:51 2012 +0900
@@ -3472,10 +3472,14 @@
 	(*mb_char2bytes)(c, buf);
 	buf[cc] = NUL;
 	ins_char_bytes(buf, cc);
+	AppendToRedobuff(buf);
     }
     else
 #endif
+    {
 	ins_char(c);
+	AppendCharToRedobuff(c);
+    }
 
     /* If we didn't complete finding matches we must search again. */
     if (ins_compl_need_restart())
