diff -r 27bfb1674333 src/Make_mvc.mak
--- a/src/Make_mvc.mak	Wed Dec 12 18:20:32 2012 +0100
+++ b/src/Make_mvc.mak	Wed Dec 12 13:44:44 2012 -0500
@@ -466,16 +466,22 @@ OPTFLAG = /Ox
 
 !if ("$(MSVCVER)" == "8.0") || ("$(MSVCVER)" == "9.0") || ("$(MSVCVER)" == "10.0") || ("$(MSVCVER)" == "11.0")
 # Use link time code generation if not worried about size
 !if "$(OPTIMIZE)" != "SPACE"
 OPTFLAG = $(OPTFLAG) /GL
 !endif
 !endif
 
+!if ("$(MSVCVER)" == "11.0")
+!if "$(OPTIMIZE)" == "MAXSPEED"
+OPTFLAG = $(OPTFLAG) /Qpar /Ob2 /Oi /Ot /Oy- /GT /GF /GS- /Gy
+!endif
+!endif
+
 # (/Wp64 is deprecated in VC9 and generates an obnoxious warning.)
 !if ("$(MSVCVER)" == "7.0") || ("$(MSVCVER)" == "7.1") || ("$(MSVCVER)" == "8.0") 
 CFLAGS=$(CFLAGS) $(WP64CHECK)
 !endif
 
 CFLAGS = $(CFLAGS) $(OPTFLAG) -DNDEBUG $(CPUARG)
 RCFLAGS = $(rcflags) $(rcvars) -DNDEBUG
 ! ifdef USE_MSVCRT
